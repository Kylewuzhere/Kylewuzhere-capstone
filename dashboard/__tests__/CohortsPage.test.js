import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Cohorts from "../src/app/dashboard/cohorts/page.js";

beforeEach(async () => {
  render(<Cohorts />);
});

describe("Cohorts", () => {
  it("renders Cohorts correctly", async () => {
    const tree = renderer.create(<Cohorts />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders a list of cohorts", async () => {
    const cohorts = await screen.findAllByRole("link");
    expect(cohorts).toHaveLength(9);
  });
});
