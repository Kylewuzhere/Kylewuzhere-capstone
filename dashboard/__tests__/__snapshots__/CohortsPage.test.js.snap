// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohorts renders Cohorts correctly 1`] = `
<div
  className="w-full h-full bg-grey-light"
>
  <div
    className="w-full mb-2 border-b-2 border-black"
  >
    <div
      className="w-full h-12 bg-grey-lighter text-black text-3xl font-semibold pt-11 pb-9 px-12 flex items-center justify-between"
    >
      <h1
        className="ml-4"
      >
        COHORTS
      </h1>
      <div
        className="text-sm flex space-x-4 mr-4"
      >
        <button
          className="rounded border border-black bg-gray-200 text-white px-10 py-2"
          onClick={[Function]}
        >
          Active
        </button>
        <button
          className="rounded border border-black bg-grey-light text-white px-10 py-2"
          onClick={[Function]}
        >
          Inactive
        </button>
        <button
          className="rounded border border-black bg-grey-light text-white px-10 py-2"
          onClick={[Function]}
        >
          All
        </button>
      </div>
    </div>
  </div>
  <div
    className="mx-auto max-w-screen-md mt-10"
  >
    <div
      className="grid grid-cols-1 sm:grid-cols-2 gap-3"
    />
  </div>
</div>
`;
